src/nvim/event/process.c

```diff
@@ -269,15 +273,18 @@ static void children_kill_cb(uv_timer_t *handle)
     if (!proc->stopped_time) {
       continue;
     }
-    uint64_t elapsed = now - proc->stopped_time;
-
-    if (!proc->term_sent && elapsed >= TERM_TIMEOUT) {
-      ILOG("Sending SIGTERM to pid %d", proc->pid);
-      uv_kill(proc->pid, SIGTERM);
-      proc->term_sent = true;
-    } else if (elapsed >= KILL_TIMEOUT) {
-      ILOG("Sending SIGKILL to pid %d", proc->pid);
-      uv_kill(proc->pid, SIGKILL);
+    uint64_t elapsed = (now - proc->stopped_time) / 1000000 + 1;
+    ILOG("elapsed: %d", elapsed);
+
+    if (elapsed >= KILL_TIMEOUT_MS) {
+      if (!proc->term_sent) {
+        ILOG("Sending SIGTERM to pid %d", proc->pid);
+        uv_kill(proc->pid, SIGTERM);
+        proc->term_sent = true;
+      } else if (elapsed >= KILL_TIMEOUT_MS * (proc->type == kProcessTypePty)) {
```

Comment:
Not necessarily, if there are several pty jobs waiting to be killed: the `repeat` for the timer is `KILL_TIMEOUT_MS`, but if new jobs are going to be queued while the timer has not been removed, this might interfere - or at least that is what I was thinking.